account_banking_pain_base/models/account_payment_order.py

```diff
@@ -522,21 +526,13 @@ def generate_remittance_info_block(self, parent_node, line, gen_args):
                     creditor_ref_info_type_or = etree.SubElement(
                         creditor_ref_info_type, 'CdOrPrtry')
                     creditor_ref_info_type_code = etree.SubElement(
-                        creditor_ref_info_type_or, 'Cd')
-                    creditor_ref_info_type_code.text = 'SCOR'
-                    creditor_ref_info_type_issuer = etree.SubElement(
-                        creditor_ref_info_type, 'Issr')
-                    creditor_ref_info_type_issuer.text = \
-                        line.communication_type
+                        creditor_ref_info_type_or, 'Prtry')
+                    creditor_ref_info_type_code.text = 'ESR'
 
                 creditor_reference = etree.SubElement(
                     creditor_ref_information, 'Ref')
 
-            creditor_reference.text = \
-                self._prepare_field(
-                    'Creditor Structured Reference',
-                    'line.communication', {'line': line}, 35,
-                    gen_args=gen_args)
+            creditor_reference.text = line.payment_line_ids[0].communication
```

Comment:
I'm not sure that this won't be problematic for other pain formats ? Why do you change this ?